src/pkgcore/ebuild/cpv.py

```diff
@@ -42,7 +42,12 @@ def isvalid_pkg_name(chunks):
     # chunk, i.e. at least one hyphen
     if len(chunks) == 1:
         return True
-    return not (isvalid_version_re.match(chunks[-1]) or isvalid_rev(chunks[-1]))
+    if isvalid_version_re.match(chunks[-1]):
```

Comment:
I kind of hate the logic flow here, even if it captures it correctly.  Suggestions welcome, just keep in mind this is something of a 'fast path' for package name validation, so it's best to align the checks accordingly.